examples/src/createReactAgent/accessExternalContextInTools.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * Access External Context in Tools
+ *
+ * This enables tools to access external application state, user data, and environmental context when executing their functions, making tool behavior context-aware and personalized.
+ *
+ * Why this is important:
+ * - Contextual Tool Execution:
+ *   Tools can adapt their behavior based on user identity, preferences, or current application state
+ * - Data Integration:
+ *   Seamlessly incorporates external data sources and APIs into tool functionality
+ * - Security and Personalization:
+ *   Ensures tools operate with appropriate permissions and user-specific context
+ *
+ * Example Scenario:
+ * You're building a travel planning assistant. When the weather tool is called, it not only gets weather data but
```

Comment:
this is probably fine, but i think looking up user specific data (eg a users purchases, by passing through their user id) is way better

this also ties into auth around tools - super important!